dev/NumberBox/NumberBox.cpp

```diff
@@ -228,6 +229,12 @@ void NumberBox::OnApplyTemplate()
     }
 }
 
+void NumberBox::OnLoaded(winrt::IInspectable const&, winrt::RoutedEventArgs const&)
+{
+    // This is done OnLoaded so TextBox VisualStates can be updated properly.
+    UpdateSpinButtonPlacement();
```

Comment:
>UpdateSpinButtonPlacement [](http://example.com/codeflow?start=4&length=25)

I'm trying to reason about why this fixes the issue... I think it is probably indicative of us needing to be attached to some layout property on a template child..